persist/fs/msgpack/encoder.go

```diff
@@ -198,13 +202,27 @@ func (enc *Encoder) encodeIndexBloomFilterInfo(info schema.IndexBloomFilterInfo)
 	enc.encodeVarintFn(info.NumHashesK)
 }
 
-func (enc *Encoder) encodeIndexEntry(entry schema.IndexEntry) {
+// We only keep this method around for the sake of testing
+// backwards-compatbility
+func (enc *Encoder) encodeIndexEntryV1(entry schema.IndexEntry) {
+	// Manually encode num fields for testing purposes
+	enc.encodeArrayLenFn(minNumIndexEntryFields)
```

Comment:
I wonder if you should just hard-code the number of use a v1-specific constant here instead of relying on the minimum (which I think is designed for decoding purposes). Seems like a safe bet since this function is designed to simulate the behavior of old binaries

Reply:
Sure thing.